File: demo_oracle_db/src/main/java/com/example/demo_oracle_db/service/product/request/ProductFilter.java
package com.example.demo_oracle_db.service.product.request;

import com.example.demo_oracle_db.validate.CustomDateConstraint;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ProductFilter {
    private String name;
    private Integer yearMaking;
    @CustomDateConstraint(pattern = "yyyy-MM-dd", message = "Start date must be format yyyy-MM-dd")
    private String startExpireDate;
    @CustomDateConstraint(pattern = "yyyy-MM-dd", message = "End date must be format yyyy-MM-dd")
    private String endExpireDate;
    private Integer minQuantity;
    private Integer maxQuantity;
    private Double minPrice;
    private Double maxPrice;
    private Integer pageNum;
    private Integer sizePage;
    private String orderCol;
    private Boolean sortDesc;

    private Long categoryId;
    private Long supplierId;
    private Long customerId;
    private String voucherCode;
}

File: demo_angular/src/app/model/dto/product-filter.ts
export class ProductFilter {
  name: string | null = null;
  yearMaking: number | null = null;

  minPrice: number | null = null;
  maxPrice: number | null = null;
  minQuantity: number | null = null;
  maxQuantity: number | null = null;
  startExpireDate: Date | null = null;
  endExpireDate: Date | null = null;

  categoryId: number | null = null;
  supplierId: number | null = null;
  voucherCode: string | null = null;
  customerId: number | null = null;

  //Order
    orderCol: string | null = null;
    sortDesc: boolean = false;
  //Paging
  pageNum: number = 1;
  sizePage: number = 8;

}
